cmd/internal/find/primary/meta/stringPredicate_test.go

```diff
@@ -0,0 +1,42 @@
+package meta
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type StringPredicateTestSuite struct {
+	ParserTestSuite
+}
+
+func (suite *StringPredicateTestSuite) TestErrors() {
+	suite.runTestCases(
+		nPETC("", "expected a nonempty string", true),
+		nPETC("-a", "-a begins with a '-'", true),
+	)
+
+	_, _, err := parseStringPredicate([]string{""})
+	suite.Regexp("expected a nonempty string", err)
+}
+
+func (suite *StringPredicateTestSuite) TestValidInput() {
+	// Test the happy cases first
+	suite.runTestCases(
+		nPTC("foo -size", "-size", "foo"),
+	)
+
+	// Now test that the predicate returns false for a non-string
+	// value or if value != s
+	p, _, err := parseStringPredicate(toTks("foo"))
+	if suite.NoError(err) {
+		suite.False(p(200))
```

Comment:
I'd like to also see a 'true' test.

Reply:
That should be addressed on line 26. `nPTC` is `(<input>, <rem>, <trueV>)`.

Reply:
Ah, I guess that works. I haven't figured out how to read some of the inputs yet. Why is `-size` in so many of them?

Reply:
Just to assert that the parse functions return the rest of `tokens`. Seemed easy enough to to test.